### ISSUE TITLE
HTTPS

### ISSUE THREAD

**Issue by witten** [mueslo/openwrt_hass_devicetracker#11]
As far as I can tell, the current `functions.sh` `post()` function unconditionally hits the `$hass_host` with unencrypted HTTP, because the code does not specify the protocol. It would be great to optionally support encrypted HTTPS as well, because HTTPS is [a valid configuration](https://www.home-assistant.io/docs/ecosystem/certificates/tls_self_signed_certificate/) for even local Home Assistant installations, and certainly for [remote-accessible](https://www.home-assistant.io/blog/2017/09/27/effortless-encryption-with-lets-encrypt-and-duckdns/) ones.

One idea is for "HTTP versus HTTPS" to just be a `openwrt_hass_devicetracker` configuration option, and then it could be passed to `curl` accordingly. However, a complicating aspect is that if the Home Assistant instance is using a self-signed cert, then `curl` may need a `--cacert` parameter with a valid Certificate Authority cert file, or at worst, `--insecure`.

**Comment by mueslo** [comment]
Does setting your `host` configuration variable to begin with `https://` not work?

**Comment by witten** [comment]
Good idea. That works! If this is the desired solution, then maybe this could just be changed into a documentation ticket, as others may also not be aware that a protocol can be included with the `host` option. Thanks.

**Comment by witten** [comment]
Thanks!

**Comment by xorbital** [comment]
Sadly, self-signed certificates result in an error.
Adding -k after "curl" in functions.sh works, so maybe a config setting would be a good solution?
